### PR TITLE
fix rocBLAS extensions API issue; add batched- and strided_batched- cases

### PR DESCRIPTION
**Pull request by mindest**
### Description
For rocBLAS extensions API:
* fix `alpha`/`beta` dtype mismatch in `rocblas_gemm_ex()`, which should be the same as `compute_type`.
* add support for `BatchedGemm` and `StridedBatchedGemm` cases.

### CODE REVIEW

File: onnxruntime/core/providers/rocm/tunable/gemm_rocblas.h [hunk 1]
```diff
@@ -103,6 +103,25 @@ constexpr rocblas_datatype RocBlasComputeTypeFor<BFloat16>(const BFloat16*) {
   return rocblas_datatype_f32_r;
 }
 
+template <typename T>
+auto maybe_cast(const T fp) {
```
Comment (cloudhan) [inline onnxruntime/core/providers/rocm/tunable/gemm_rocblas.h:107]: Naming stype does not conform to our style guideline.

File: onnxruntime/core/providers/rocm/tunable/gemm_rocblas.h [hunk 2]
```diff
@@ -169,16 +190,18 @@ class RocBlasGemmTunableOp : public TunableOp<GemmParams<T>> {
  private:
   std::vector<int> GetSolutions(const GemmParams<T>* params) {
     int num_solutions = 0;
+    auto h_a = DoCastForHalfOrBfloat16(params->alpha);
```
Comment (abudup) [inline onnxruntime/core/providers/rocm/tunable/gemm_rocblas.h:193]: Minor, here and elsewhere, consider inlining the definitions of `h_a` and `h_b`. I feel that it is more readable that way.

**Review by abudup — APPROVED** [review]
LGTM. Thanks!

**Comment by mindest** [comment]
Thanks! @abudup @cloudhan

**Pull request merged**